File: Projeto_6/src/funcionario_administrativo.py
from usuario_autenticavel import UsuarioAutenticavel
from relatorio import Relatorio
from tipo_usuario import TipoUsuario

class FuncionarioAdministrativo(UsuarioAutenticavel, Relatorio):
    def __init__(self, nome, login, senha, departamento):
        super().__init__(nome, login, senha, TipoUsuario.FUNCIONARIO_ADMINISTRATIVO)
        self.departamento = departamento

    def autenticar(self, senha: str) -> bool:
        return self._verificar_senha(senha)

    def gerar_relatorio(self):
        print(f"\nRelatório do Funcionário Administrativo: {self.nome}")
        print(f"Departamento: {self.departamento}")

File: Projeto_6/src/funcionarioAdministrativo.cpp
#include "../include/funcionarioaAdministrativo.h"
#include <iostream>

// Construtor padrão
FuncionarioAdministrativo::FuncionarioAdministrativo()
    : UsuarioAutenticavel("", "", TipoUsuario::FUNCIONARIO_ADMINISTRATIVO, ""), departamento(""), cargo("") {}

// Construtor parametrizado
FuncionarioAdministrativo::FuncionarioAdministrativo(const std::string nome, const std::string email, TipoUsuario tipo, std::string senha, const std::string departamento, const std::string cargo)
    : UsuarioAutenticavel(nome, email, TipoUsuario::FUNCIONARIO_ADMINISTRATIVO, senha), departamento(departamento), cargo(cargo) {}

// Sobrescrita de gerarRelatorio
void FuncionarioAdministrativo::gerarRelatorio() const {
    std::cout << "Funcionario Administrativo: " << nome << "\n"
              << "Departamento: " << departamento << "\n"
              << "Cargo: " << cargo << "\n";
}


bool FuncionarioAdministrativo::autenticar(const std::string& senha) const {
    return senha == getSenha(); // Usa o getter para acessar a senha armazenada
}